src/thread_pool.rs

```diff
@@ -109,14 +113,18 @@ impl Registry {
         self.work_available.notify_all();
     }
 
-    pub unsafe fn inject(&self, injected_jobs: &[*mut Job<LockLatch>]) {
+    pub unsafe fn inject(&self, injected_jobs: &[*mut (Job<LockLatch> + 'static)]) {
```

Comment:
Yes, it fails to compile otherwise. It's automatically added by lifetime inference in other places, but it needs to be explicitly specified here.